lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,11 +166,65 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
+
+    // Add Hash.
+    if (ppResult != nullptr && *ppResult != nullptr) {
+      HRESULT HR;
+      (*ppResult)->GetStatus(&HR);
+      if (SUCCEEDED(HR)) {
+        CComPtr<IDxcBlob> pObject;
+        HR = (*ppResult)->GetResult(&pObject);
+        if (SUCCEEDED(HR)) {
+          LPVOID PTR = pObject->GetBufferPointer();
+          if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
+            HashAndUpdate((DxilContainerHeader *)PTR);
+        }
+      }
+    }
     return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
 }
 
+// Try hashing the source contained in ContainerHeader using retail and debug
```

Comment:
I'm confused by the use of "source" here.  This doesn't refer to HLSL source code does it?  Also, it looks like it hashes the data stored in the container, not in the ContainerHeader.

The commend helps explain this function a bit more, but the name "FindHashFunctionFromSource" could be more descriptive.

Reply:
What if it were:

```c++
static HASH_FUNCTION_PROTO* DetermineHashFunctionFromContainerContents(
    const DxilContinerHeader* ContainerHeader) {
...
}
```